### ISSUE TITLE
`NIOLock` missing `withLockVoid(_:)`

### ISSUE THREAD

**Issue by MahdiBM** [apple/swift-nio#2275]
`Lock` has been recently deprecated in favor of the new `NIOLock`.
The warnings mention that "`Lock` has been renamed to `NIOLock`". Considering `Lock` had `withLockVoid(_:)`, I would expected `NIOLock` to have that function as well, but it's missing.

I'm not sure if this is considered "api breaking" or not, but it was an unexpected change to me since the warnings claimed that this is only a "rename".

**Comment by Lukasa** [comment]
Yeah, this isn't an API break but it's an unfortunate oversight. Would you like to open a PR to add the function?

**Comment by MahdiBM** [comment]
Surely. Thanks for the answer.

**Comment by adam-fowler** [comment]
Yeah just found this when moving across to NIOLock in Soto
